services/opentsdb/service.go

```diff
@@ -408,20 +455,21 @@ func (s *Service) serveHTTP() {
 
 // processBatches continually drains the given batcher and writes the batches to the database.
 func (s *Service) processBatches(batcher *tsdb.PointBatcher) {
-	defer s.wg.Done()
 	for {
 		select {
+		case <-s.done:
+			return
 		case batch := <-batcher.Out():
+			// Will attempt to create database if not yet created.
+			s.createInternalStorage()
```

Comment:
Since there is nothing we can do but log the error, it seems cleaner to have `createInternalStorage` check for an error and log it itself.